sdk/vision/azure-ai-vision-imageanalysis/README.md

```diff
@@ -0,0 +1,338 @@
+
+Notes:
+* Caption is only available in some Azure regions. See [Prerequisites](#prerequisites).
+* Caption is only supported in English at the moment.
+
+<!-- SNIPPET:sample_caption_image_file.caption -->
+
+```python
+# Load image to analyze into a 'bytes' object
+with open("sample.jpg", 'rb') as f:
+    image_buffer = bytes(f.read())
+
+# Get a caption for the image. This will be a synchronously (blocking) call.
+result = client.analyze(
+    image_content = image_buffer,
+    visual_features = [ VisualFeatures.CAPTION ],
+    gender_neutral_caption = True) # Optional (default is False)
+
+client.close()
+
+# Print caption results to the console
+print(f"Image analysis results:")
+print(f" Caption:")
+if result.caption is not None:
+    print(f"   '{result.caption.text}', Confidence {result.caption.confidence:.4f}")
+```
+
+<!-- END SNIPPET -->
+
+To generate captions for additional images, simply call `analyze` multiple times. You can use the same `ImageAnalysisClient` do to multiple analysis calls.
+
+### Generate an image caption for an image URL
+
+This example is similar to the above, expect it calls the `analyze` method and provides a [publicly accessible image URL](https://aka.ms/azai/vision/image-analysis-sample.jpg) instead of a file name.
+
+<!-- SNIPPET:sample_caption_image_url.caption -->
+
+```python
+# Get a caption for the image. This will be a synchronously (blocking) call.
+result = client.analyze(
+    image_content = "https://aka.ms/azai/vision/image-analysis-sample.jpg",
+    visual_features = [ VisualFeatures.CAPTION ],
+    gender_neutral_caption = True) # Optional (default is False)
+
+client.close()
+
+# Print caption results to the console
+print(f"Image analysis results:")
+print(f" Caption:")
+if result.caption is not None:
+    print(f"   '{result.caption.text}', Confidence {result.caption.confidence:.4f}")
+```
+
+<!-- END SNIPPET -->
+
+### Extract text from an image file
+
+This example demonstrates how to extract printed or hand-written text for the image file `sample.jpg` using the `ImageAnalysisClient`. The synchronous (blocking) `analyze` method call returns an `ImageAnalysisResult` object with a `read` property of type `ReadResult`. It includes a list of text lines and a bounding polygon surrounding each text line. For each line, it also returns a list of words in the text line and a bounding polygon surrounding each word.
+
+<!-- SNIPPET:sample_ocr_image_file.read -->
+
+```python
+# Load image to analyze into a 'bytes' object
+with open("sample.jpg", 'rb') as f:
+    image_buffer = bytes(f.read())
+
+# Extract text (OCR) from an image stream. This will be a synchronously (blocking) call.
+result = client.analyze(
+    image_content = image_buffer,
+    visual_features = [ VisualFeatures.READ ])
+
+client.close()
```

Comment:
Is `client.close()` in the samples necessary? @lmazuel @iscai-msft

Reply:
I added it because when I run the async sample (see [\samples\async_samples\sample_caption_image_file_async.py](https://github.com/Azure/azure-sdk-for-python/blob/dargilco/image-analysis-python/sdk/vision/azure-ai-vision-imageanalysis/samples/async_samples/sample_caption_image_file_async.py) in this PR) without calling `await client.close()` at the end, I get this console output:
```
Image analysis results:
 Caption:
   'a woman wearing a mask sitting at a table with a laptop', Confidence 0.8498
 Image height: 576
 Image width: 864
 Model version: 2023-10-01
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x000002477FC5B070>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x000002477FC67220>, 814467.359)]']
connector: <aiohttp.connector.TCPConnector object at 0x000002477FC5AF50>
```

When I add `await client.close()` the `Unclosed client session` line and below are no longer shown:

```
Image analysis results:
 Caption:
   'a woman wearing a mask sitting at a table with a laptop', Confidence 0.8498
 Image height: 576
 Image width: 864
 Model version: 2023-10-01
```

So, it looks like it is required for async samples, but may not be required for sync samples. I don't see a similar `Unclosed client session` in the sync samples when I remove ` client.close()`. 

Is this expected?

Searching in other `sdk\**\samples\**` folders, I see many examples of `await client.close()` being used, but very few `client.close()`.

So perhaps I should also remove `client.close()` from the sync samples. Just need to make sure the need for `await client.close()` is clearly called out in the README docs when using async pattern.

Reply:
@msyyc I made another commit to remove the `client.close()` from the sync samples. I left `await client.close()` in the async samples. Thanks!